test/runner/moc_test.ml

```diff
@@ -0,0 +1,190 @@
+          let mo_files = collect_drun_mo_files drun_file_path in
+          Printf.printf "Drun .mo files: ";
+          print_list (List.of_seq (StringSet.to_seq mo_files));
+
+          (* Compile .mo files *)
+          StringSet.iter
+            (fun mo_file ->
+              (* TODO: ../run-drun part should be gone *)
+              let mo_file_path = Printf.sprintf "../run-drun/%s" mo_file in
+              Printf.printf "Compiling mo file: %s\n" mo_file_path;
+
+              let mo_base =
+                Filename.remove_extension (Filename.basename mo_file)
+              in
+              let moc_cmd =
+                Printf.sprintf "moc --hide-warnings -c %s -o %s/%s.drun.wasm\n"
+                  mo_file_path out_dir mo_base
+              in
+
+              Printf.printf "Compiling %s: `%s`" mo_file moc_cmd;
+
+              let mkdir_exit =
+                Sys.command (Printf.sprintf "mkdir -p %s" out_dir)
+              in
+              Alcotest.(check int) "mkdir exit code" 0 mkdir_exit;
+
+              let moc_exit = Sys.command moc_cmd in
+              Alcotest.(check int) "moc exit code" 0 moc_exit)
+            mo_files;
+
+          (* Run drun, compare outputs *)
+          let drun_script =
+            prepare_drun_script drun_file_path out_dir test_name "drun"
+          in
+
+          Printf.printf "Drun script: %s\n" drun_script;
+
+          let drun_processed_script_path =
+            Printf.sprintf "%s/%s.drun.drun" out_dir test_name
+          in
+          let drun_out = open_out drun_processed_script_path in
+          Printf.fprintf drun_out "%s\n" drun_script;
+          close_out drun_out;
+
+          let actual_drun_output_path =
+            Printf.sprintf "%s/%s.drun.out" out_dir test_name
+          in
+          let drun_cmd =
+            Printf.sprintf "drun -c %s --extra-batches 1 %s 2>&1 > %s"
+              drun_config_path drun_processed_script_path
+              actual_drun_output_path
+          in
+
+          Printf.printf "Running %s\n" drun_cmd;
+
+          let drun_exit = Sys.command drun_cmd in
+          Alcotest.(check int) "drun exit code" 0 drun_exit;
+
+          let expected_drun_output_path =
+            Printf.sprintf "../run-drun/ok/%s.drun.ok" test_name
+          in
+
+          let expected_output = read_file expected_drun_output_path in
+          let actual_output = read_file actual_drun_output_path in
+
+          if expected_output <> actual_output then (
+            let diff_cmd =
+              Printf.sprintf
+                "diff -a -u -N --label expected ../run-drun/ok/%s.drun.ok \
+                 --label actual %s/%s.drun.out"
+                test_name out_dir test_name
+            in
+
+            (* TODO: Somehow this line is printed after the `diff` output.
+               OCaml runtime probably has its own output buffer, we need to
+               flush it somehow. *)
+            Printf.printf "Outputs do not match, running %s\n" diff_cmd;
```

Comment:
You can use format string `%!` to flush, I think https://ocaml.org/releases/4.07/htmlman/libref/Printf.html.

Reminds me of something I heard in school: "flush twice, it's a long way to the candy machine."

Reply:
You're a star for doing this. What do we conclude if the ocaml is way longer than the bash? ;->

Reply:
> What do we conclude if the ocaml is way longer than the bash?

I don't think people dislike bash because of its verbosity...